sdclientapi/timestamps.py

```diff
@@ -0,0 +1,28 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# Copyright © 2022 The Freedom of the Press Foundation.
+from datetime import datetime
+from typing import Optional
+
+
+def parse(date_string: str) -> Optional[datetime]:
+    """Parse date strings in a few historical formats."""
+    dt = None  # type: Optional[datetime]
```

Comment:
nit: you can remove this assignment because the it's not used as a fallback anywhere, and also return straight from the datetime calls, like:
```
def parse(date_string: str) -> Optional[datetime]:
    """Parse date strings in a few historical formats."""
    try:
        # ISO8061 and RFC3339
        # '2022-02-09T07:45:26.082728+00:00'
        return datetime.fromisoformat(date_string)  # type: ignore
    except ValueError:
        try:
            # '2022-02-09T07:45:26.082728Z'
            # DEPRECATED: Please remove this code once the server doesn't require it anymore.
            return datetime.strptime(date_string, "%Y-%m-%dT%H:%M:%S.%f%z")
        except ValueError:
            try:
                # '2022-02-09T07:45:26.082728+00:00Z' invalid format, but we support it.
                # DEPRECATED: Please remove this code once the server doesn't require it anymore.
                return datetime.strptime(date_string, "%Y-%m-%dT%H:%M:%S.%f%zZ")
            except ValueError:
                return None
```

Reply:
:100: I had added it because at some point the type of the return value of `datetime.fromisodate()` seemed to be `Any`, but I can't reproduce that, so yes, absolutely!